frank/utilities.py

```diff
@@ -231,58 +324,90 @@ def cut_data_by_baseline(u, v, vis, weights, cut_range):
 
     return u_cut, v_cut, vis_cut, weights_cut
 
-
-def apply_correction_to_weights(u, v, ReV, weights, nbins=300):
+def estimate_weights(u, v, V, nbins=300, log=True, use_median=False):
     r"""
-    Estimate and apply a correction factor to the data's weights by comparing
-    binnings of the real component of the visibilities under different
-    weightings. This is useful for mock datasets in which the weights are all
-    unity.
+    Estimate the weights using the variance of the binned visibilities.
+
+    The estimation is done assuming that the variation in each bin is dominated
+    by the noise. This will be true if:
+        1) The source is axi-symmetric,
+        2) The uv-points have been deprojected,
+        3) The bins are not too wide,
+    Otherwise the variance may be dominated by real variations in the
```

Comment:
`real` is ambiguous...
```suggestion
    Otherwise the variance may be dominated by intrinsic variations in the
```